src/oas.md

```diff
@@ -44,6 +44,28 @@ Each template expression in the path MUST correspond to a path parameter that is
 
 The value for these path parameters MUST NOT contain any unescaped "generic syntax" characters described by [RFC3986](https://tools.ietf.org/html/rfc3986#section-3): forward slashes (`/`), question marks (`?`), or hashes (`#`).
 
+The path templating expression is defined by the following [ABNF](https://tools.ietf.org/html/rfc5234) syntax
+
+```abnf
+; OpenAPI Path Templating ABNF syntax
+path-template                  = slash *( path-segment slash ) [ path-segment ]
+path-segment                   = 1*( path-literal / template-expression )
+slash                          = "/"
```

Comment:
```suggestion
path-template                  = "/" [ path-template-nz ]
path-template-nz               = path-segment *( "/" path-segment )
path-segment                   = 1*( path-literal / template-expression )
```

(porting the suggestion from @char0n so we can have a focused discussion)

@char0n I think one of the challenges of the syntax you're proposing is that it doesn't allow a trailing slash `/foo/bar/` where the current one does.

I don't have a strong opinion on the slash being a terminal or not.

Reply:
> @char0n I think one of the challenges of the syntax you're proposing is that it doesn't allow a trailing slash /foo/bar/ where the current one does.

It should allow, as per RFC 3986. This ABNF MUST correctly reflect that.

I've updated to reflect that as well - https://github.com/OAI/OpenAPI-Specification/pull/4244#issuecomment-2545553855

Reply:
just to confirm I'm not missing anything, the only difference is the slash not being a terminal, correct?

Reply:
sorry, I miss-used the terminology here. What I meant was: can you confirm this is the only difference remaining please?

```suggestion
path-template                  = "/" *( path-segment "/" ) [ path-segment ]
path-segment                   = 1*( path-literal / template-expression )
```

Reply:
Yes I confirm. These are the only last changes needed.